core/ErrorQueue.cc

```diff
@@ -186,4 +190,16 @@ bool ErrorQueue::queueIsEmptyApprox() const {
     return this->queue.sizeEstimate() == 0;
 }
 
+bool ErrorQueue::hasFlushesFor(core::FileRef file) const {
```

Comment:
I think we just need to check if error queue has marked _any_ file for flushing prior to typecheck.

You can do this in `pipeline::typecheck` before any typechecking begins.